h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -439,23 +455,23 @@ private void reconcileMaxRuntime(Key<Grid<MP>> gridKey, Model.Parameters params)
       Log.info("Due to the grid time limit, changing model max runtime to: " + params._max_runtime_secs + " secs.");
     }
   }
-
+  
+  private void beforeGridStart(Grid grid) {
+    if (_exportFrames) {
+      attemptGridSave(grid);
+      new Recovery<Grid>(_hyperSpaceWalker.getParams()._export_checkpoints_dir).exportReferences(grid);
```

Comment:
@honzasterba IMHO this is mixing two concepts together

Parameter `export_checkpoints_dir` is meant for the _user_ to export models - this can have a different motivation than just recovery.

Fault-tolerance is driven by the software, not the user. I think there should be 2 locations - one for user checkpointed objects (that one is preserved after modeling ends) and one for fault-tolerance objects (that was is cleaned up automatically).